src/controller/CHIPDevice.cpp

```diff
@@ -171,15 +170,16 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
     serializable.mAdminId    = Encoding::LittleEndian::HostSwap16(mAdminId);
 
     Transport::PeerConnectionState * connectionState = mSessionManager->GetPeerConnectionState(mSecureSession);
-    VerifyOrExit(connectionState != nullptr, error = CHIP_ERROR_INCORRECT_STATE);
-    localMessageCounter = connectionState->GetSessionMessageCounter().GetLocalMessageCounter().Value();
-    peerMessageCounter  = connectionState->GetSessionMessageCounter().GetPeerMessageCounter().GetCounter();
+    if (connectionState != nullptr)
+    {
+        localMessageCounter = connectionState->GetSessionMessageCounter().GetLocalMessageCounter().Value();
+        peerMessageCounter  = connectionState->GetSessionMessageCounter().GetPeerMessageCounter().GetCounter();
+    }
```

Comment:
why is it now ok to not have a connection state?
A comment and explicit init on  the else branch would help: if I see the else branch setting counters to 0, I avoid the need to scroll up to read the code and also ensure nobody else changes the counter values from their initial value.

Maybe even:

```cpp
if (connectionState != nullptr)
    serializable.mLocalMessageCounter = 
        connectionState->GetSessionMessageCounter().GetLocalMessageCounter().Value();	
    serializable.mPeerMessageCounter = 
        connectionState->GetSessionMessageCounter().GetPeerMessageCounter().GetCounter();		       
} else {
    serializable.mLocalMessageCounter  = 0;
    serializable.mPeerMessageCounter = 0;
}
serializable.mLocalMessageCounter  = Encoding::LittleEndian::HostSwap32(mLocalMessageCounter);
serializable.mPeerMessageCounter = Encoding::LittleEndian::HostSwap32(serializable.mPeerMessageCounter);
```

and drop those local variables.